### PR TITLE
Disallow non-alphanumeric in `HeaderName::from_str`

### PR DESCRIPTION
**Pull request by caspervonb**
Closes https://github.com/nats-io/nats.rs/issues/941

### CODE REVIEW

File: async-nats/src/header.rs [hunk 1]
```diff
@@ -309,6 +309,10 @@ impl FromStr for HeaderName {
     type Err = ParseError;
 
     fn from_str(s: &str) -> Result<Self, Self::Err> {
+        if s.contains(|c: char| !c.is_ascii_alphanumeric() && c != '-') {
```
Comment (Jarema) [inline async-nats/src/header.rs:312]: If we want to use this error, it really should return some context why parsing failed.

**Review by Jarema — CHANGES_REQUESTED** [review]

File: async-nats/src/header.rs (hunk 1 above)
Comment (caspervonb) [reply async-nats/src/header.rs:312]: Error type could be changed into a `ParseHeaderNameError` and vice versa for header values, but follow up imo to unblock the rest of the queue.

**Review by Jarema — APPROVED** [review]
LGTM!

**Review by Jarema — APPROVED** [review]
LGTM!

**Pull request merged**